### PR TITLE
Zh tn 0712

### PR DESCRIPTION
**Pull request by BuyuanCui**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

# Before your PR is "Ready for review"
**Pre checks**:
- [*] Have you signed your commits? Use ``git commit -s`` to sign.
- [*] Do all unittests finish successfully before sending PR?
   1) ``pytest`` or (if your machine does not have GPU) ``pytest --cpu`` from the root folder (given you marked your test cases accordingly `@pytest.mark.run_only_on('CPU')`).
   2) Sparrowhawk tests ``bash tools/text_processing_deployment/export_grammars.sh --MODE=test ...``
- [*] If you are adding a new feature: Have you added test cases for both `pytest` and Sparrowhawk [here](tests/nemo_text_processing).
- [*] Have you added ``__init__.py`` for every folder and subfolder, including `data` folder which has .TSV files?
- [*] Have you followed codeQL results and removed unused variables and imports (report is at the bottom of the PR in github review box) ?
- [*] Have you added the correct license header `Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.` to all newly added Python files?
- [*] If you copied [nemo_text_processing/text_normalization/en/graph_utils.py](nemo_text_processing/text_normalization/en/graph_utils.py) your header's second line should be `Copyright 2015 and onwards Google, Inc.`. See an example [here](https://github.com/NVIDIA/NeMo-text-processing/blob/main/nemo_text_processing/text_normalization/en/graph_utils.py#L2).
- [*] Remove import guards (`try import: ... except: ...`) if not already done.
- [*] If you added a new language or a new feature please update the [NeMo documentation](https://github.com/NVIDIA/NeMo/blob/main/docs/source/nlp/text_normalization/wfst/wfst_text_normalization.rst) (lives in different repo).
- [*] Have you added your language support to [tools/text_processing_deployment/pynini_export.py](tools/text_processing_deployment/pynini_export.py).
  

**PR Type**:
- [*] New Feature
- [*] Bugfix
- [ ] Documentation
- [ ] Test

If you haven't finished some of the above items you can still open "Draft" PR.

### CODE REVIEW

File: nemo_text_processing/text_normalization/zh/data/date/year_suffix.tsv [hunk 1]
```diff
@@ -1,6 +0,0 @@
-来
```
Comment (fayejf) [inline nemo_text_processing/text_normalization/zh/data/date/year_suffix.tsv:1]: why this file is deleted?

File: nemo_text_processing/text_normalization/zh/data/whitelist.tsv [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+IQ	IQ
+EQ	EQ
+cctv	CCTV
+C C t v	CCTV
+kfc	KFC
+K F C	KFC
+Steam	steam
+phd	博士
+PhD	博士
+Dr.	医生
```
Comment (fayejf) [inline nemo_text_processing/text_normalization/zh/data/whitelist.tsv:82]: How to distinguish Dr. for 医生 and 博士？

File: tests/nemo_text_processing/zh/data_text_normalization/test_cases_math.txt [hunk 3]
```diff
@@ -1,3 +1,2 @@
 78:96~七十八比九十六
--23℃~负二十三摄氏度
```
Comment (fayejf) [inline tests/nemo_text_processing/zh/data_text_normalization/test_cases_math.txt:2]: why this is deleted?

File: tests/nemo_text_processing/zh/data_text_normalization/test_cases_measure.txt [hunk 4]
```diff
@@ -2,5 +2,4 @@
 38°C~三十八摄氏度
 120m²~一百二十平方米
 10ms~十毫秒
-6.3%~百分之六点三
```
Comment (fayejf) [inline tests/nemo_text_processing/zh/data_text_normalization/test_cases_measure.txt:5]: same as above

File: tests/nemo_text_processing/zh/test_date.py [hunk 5]
```diff
@@ -30,11 +30,11 @@ def test_norm_date(self, test_input, expected):
...
-    inverse_normalizer = InverseNormalizer(lang='zh', cache_dir=CACHE_DIR, overwrite_cache=False)
+    # inverse_normalizer = InverseNormalizer(lang='zh', cache_dir=CACHE_DIR, overwrite_cache=False)
 
-    @parameterized.expand(parse_test_case_file('zh/data_inverse_text_normalization/test_cases_date.txt'))
-    @pytest.mark.run_only_on('CPU')
-    @pytest.mark.unit
-    def test_denorm(self, test_input, expected):
-        pred = self.inverse_normalizer.inverse_normalize(test_input, verbose=False)
-        assert pred == expected
+    # @parameterized.expand(parse_test_case_file('zh/data_inverse_text_normalization/test_cases_date.txt'))
```
Comment (fayejf) [inline tests/nemo_text_processing/zh/test_date.py:35]: remove if useless

File: tests/nemo_text_processing/zh/test_decimal.py [hunk 6]
```diff
@@ -14,17 +14,27 @@
...
     @pytest.mark.unit
-    def test_denorm(self, test_input, expected):
-        pred = self.inverse_normalizer.inverse_normalize(test_input, verbose=False)
-        assert pred == expected
+    def test_norm_date(self, test_input, expected):
+        preds = self.normalizer_zh.normalize(test_input)
+        assert expected == preds
+
+    # inverse_normalizer = InverseNormalizer(lang='zh', cache_dir=CACHE_DIR, overwrite_cache=False)
+
```
Comment (fayejf) [inline tests/nemo_text_processing/zh/test_decimal.py:34]: same. clean the code

File: tests/nemo_text_processing/zh/test_fraction.py [hunk 7]
```diff
@@ -30,11 +30,11 @@ def test_norm_fraction(self, test_input, expected):
...
+    # inverse_normalizer = InverseNormalizer(lang='zh', cache_dir=CACHE_DIR, overwrite_cache=False)
 
-    @parameterized.expand(parse_test_case_file('zh/data_inverse_text_normalization/test_cases_fraction.txt'))
-    @pytest.mark.run_only_on('CPU')
-    @pytest.mark.unit
-    def test_denorm(self, test_input, expected):
-        pred = self.inverse_normalizer.inverse_normalize(test_input, verbose=False)
-        assert pred == expected
+    ##@parameterized.expand(parse_test_case_file('zh/data_inverse_text_normalization/test_cases_fraction.txt'))
+    # @pytest.mark.run_only_on('CPU')
```
Comment (fayejf) [inline tests/nemo_text_processing/zh/test_fraction.py:36]: please remove all unnessary # to make code clean

**Review by fayejf — CHANGES_REQUESTED** [review]
Great PR. overall LGTM. some comments. Please remove unnecessary comments to make code clean. fix CodeQL issue. And make sure CI is on and passed

File: nemo_text_processing/text_normalization/zh/data/date/year_suffix.tsv (hunk 1 above)
Comment (BuyuanCui) [reply nemo_text_processing/text_normalization/zh/data/date/year_suffix.tsv:1]: My idea was that comparing <2012年> and <2012年来>, the inclusion on this character seemed not necessary in the later example.

File: tests/nemo_text_processing/zh/data_text_normalization/test_cases_math.txt (hunk 3 above)
Comment (BuyuanCui) [reply tests/nemo_text_processing/zh/data_text_normalization/test_cases_math.txt:2]: The removed item seems like the input/output for one of our classes, <measure>, which is accountable for width, heights, and etc. I thought temperature should also be in the <measure>'s file.

File: tests/nemo_text_processing/zh/data_text_normalization/test_cases_measure.txt (hunk 4 above)
Comment (BuyuanCui) [reply tests/nemo_text_processing/zh/data_text_normalization/test_cases_measure.txt:5]: I moved it under the <fraction>. The updated grammar <fraction> grammar can process inputs with "%".

File: tests/nemo_text_processing/zh/test_date.py (hunk 5 above)
Comment (BuyuanCui) [reply tests/nemo_text_processing/zh/test_date.py:35]: This part is responsible for running ITN test. I had separate PRs for TN and ITN, so I commented the ITN part out temporally. This part is needed.

File: tests/nemo_text_processing/zh/test_decimal.py (hunk 6 above)
Comment (BuyuanCui) [reply tests/nemo_text_processing/zh/test_decimal.py:34]: This part is responsible for running ITN test. I had separate PRs for TN and ITN, so I commented the ITN part out temporally. This part is needed.

File: tests/nemo_text_processing/zh/test_fraction.py (hunk 7 above)
Comment (BuyuanCui) [reply tests/nemo_text_processing/zh/test_fraction.py:36]: This part is responsible for running ITN test. I had separate PRs for TN and ITN, so I commented the ITN part out temporally. This part is needed.

Comment (BuyuanCui) [reply tests/nemo_text_processing/zh/test_fraction.py]: Added

File: tests/nemo_text_processing/zh/test_decimal.py (hunk 6 above)
Comment (BuyuanCui) [reply tests/nemo_text_processing/zh/test_decimal.py:34]: Added

File: tests/nemo_text_processing/zh/test_date.py (hunk 5 above)
Comment (BuyuanCui) [reply tests/nemo_text_processing/zh/test_date.py]: Added

**Review by fayejf — DISMISSED** [review]
LGTM. Let's see the test results

**Comment by BuyuanCui** [comment]
Found coding format issue. Will update and push again. 
Also, rerunning tests. Zh_TN had failure cases due to the weights applied to the grammars. Already fixed.

**Comment by BuyuanCui** [comment]
Everything updated. Local test pass.

**Review by fayejf — APPROVED** [review]
LGTM! Thanks

**Pull request merged**